ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,277 @@
+        {
+            return false;
+        }
+    }
+
+    // figure out if ellipsis is met in input subscripts
+    const std::regex ellipsis_regex("\\.\\.\\.");
+    bool is_ellipsis_met = false;
+    std::smatch ellipsis_match;
+    for (auto const& input_subscript : input_subscripts)
+    {
+        if (std::regex_search(input_subscript, ellipsis_match, ellipsis_regex) == true)
+        {
+            is_ellipsis_met = true;
+            break;
+        }
+    }
+
+    if (pos_output_delimeter == std::string::npos)
+    {
+        // recover output subscript
+        output_subscript = "";
+        for (auto const& input_subscript : input_subscripts)
+        {
+            for (auto const& label : input_subscript)
+            {
+                if (std::isalpha(label) && output_subscript.find(label) == std::string::npos)
+                {
+                    output_subscript += label;
+                }
+            }
+        }
+        std::sort(output_subscript.begin(), output_subscript.end());
+        if (is_ellipsis_met)
+        {
+            output_subscript = "..." + output_subscript;
+        }
+    }
+    else
+    {
+        output_subscript = m_equation.substr(pos_output_delimeter + 2);
+
+        // check that the output subscript has the correct format
+        if (std::regex_match(output_subscript, subscript_regex) == false)
+        {
+            return false;
+        }
+
+        // if the ellipsis is met in input subscripts, one ellipsis must be in the output subscript
+        std::regex_search(output_subscript, ellipsis_match, ellipsis_regex);
+        if (is_ellipsis_met && ellipsis_match.size() != 1)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void op::v7::Einsum::extract_labels(std::string const& subscript, std::vector<std::string>& labels)
+{
+    labels.clear();
+    auto subscript_length = subscript.length();
+    for (size_t ch_idx = 0; ch_idx < subscript_length;)
+    {
+        if (std::isalpha(subscript[ch_idx]))
+        {
+            labels.push_back(subscript.substr(ch_idx, 1));
+        }
+        else if (((subscript_length - ch_idx) > 2) &&
+                 (subscript.substr(ch_idx, 3).compare("...") == 0))
+        {
+            labels.push_back("...");
+            ch_idx += 2;
+        }
+        ++ch_idx;
+    }
+}
+
+void op::v7::Einsum::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v7_Einsum_validate_and_infer_types);
+
+    // check that Einsum operation has at least one input
+    auto num_inputs = get_input_size();
+    NODE_VALIDATION_CHECK(this, num_inputs > 0, "Einsum must have at least one input.");
+
+    // check that all inputs have the same type and the type is numeric
+    const auto& input_type_0 = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_type_0 == element::f16 || input_type_0 == element::f32,
+                          "The input type for Einsum operation must be numeric.");
```

Comment:
Here you check for floating-point types, but numeric includes integer types, so the message is misleading.

Reply:
Fixed.